### ISSUE TITLE
How to use pipeline commands properly?

### ISSUE THREAD

**Issue by omerkawaz** [sewenew/redis-plus-plus#288]
Hello,
I'm trying to implement a correct mechanism of  setting/getting with a simple pipeline object in order to maximize efficiency.
I want to write an auxiliary function that should deal with all exceptions since try/catching each time is messy.
do the pipeline set/get options throw by themselves without the exec() call? or I can call them on the pipeline object and than put just the exec() call in a try/catch block?
in other words is this ok?
```
for (int i = 0; i < 10; i++) {
    pipe.set(key[i], val[i]);
}
// assuming all this function does is handling exceptions, no heavy calculations or actions done.
functionThatCallsExecAndDealsWithExceptions(pipe);
```

Moreover, say that an exception has been thrown, what is the correct way of destroying that pipeline object?

Thanks in advance.

**Comment by sewenew** [comment]
> do the pipeline set/get options throw by themselves without the exec() call?

Yes. It might throw exception without exec(). So in your case, you need to wrap the for loop into a try block:

```
try {
    for (int i = 0; i < 10; i++) {
        pipe.set(key[i], val[i]);
    }
    auto replies = pipe.exec();
    // handle replies
} catch (const Error &e) {
    // error handling.
}
```

> what is the correct way of destroying that pipeline object?

Make the pipeline object out of scope (and the destructor will be called), or use move `operator=` to assign it with a new pipeline.

**Comment by omerkawaz** [comment]
Thank you so much for the quick response.
